behave_html_pretty_formatter/html_pretty.py

```diff
@@ -605,7 +607,44 @@ def generate_step(self, formatter, scenario_status):
                     continue
                 self.generate_embed(embed_data)
 
-    def generate_embed(self, embed_data):
+    def generate_download_button(self, embed_data, data, use_caption):
+        """
+        Creates Download button in HTML.
+
+        This should not be part of Embed class, as Embed objects are
+        returned to user for later modification of data, we want to
+        prevent accidental call of this.
+        """
+
+        def _create_download_button():
+            args = f"'embed_{embed_data.uid}','{use_caption}'"
+            onclick = f"download_embed({args})"
+            a(
+                "[Download]",
+                href="#/",
+                cls="embed_download margin-bottom",
+                onclick=onclick,
+            )
+
+        # Rule for embed_data.download_button as None - default value.
+        if embed_data.download_button is None:
+            # Do not create button if there is mime type text with less then 20 lines.
+            if "text" in embed_data.mime_type and data.count("\n") < 20:
+                return
+
+            # Do not create button if the mime type is link.
+            if "link" in embed_data.mime_type:
+                return
+
+            # In all other cases the button is valid.
+            _create_download_button()
+
+        # Rule for embed_data.download_button as True.
+        elif embed_data.download_button:
+            # Create download for all cases.
+            _create_download_button()
+
+    def generate_embed(self, embed_data):  # pylint: disable=too-many-branches
```

Comment:
Pylint actually complains rightly here: You have too many branches in the code of this function. See the [Pylint documentation](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-many-branches.html) for explanation.

I believe that (also) refers to the nested `with` statements in lines 680-688. You can write them separated by commas instead of nested, e.g.

```python
with div(cls="embed_button collapse"), \
    div(cls="link"), \
    div(cls="embed_button collapse"), \
    div(cls="link"):
    ...
```

I suggest you better try this instead of silencing Pylint. After getting used to its complaints you'll find out that – tedious as they may seem – following its suggestions makes your code better.

It still amazes me whenever this happens to me, again and again. :sparkles:

Reply:
Thank you for suggesting the separation with comas. It did not even occur to me it can be used this way. We will for sure fix this with future PR.

As for the pylint being silenced, it is clearly a band aid just to get through checks and get the changes to main branch. We plan to address the pylint issue before we make a new release.